core/src/test/java/org/infinispan/distribution/ch/impl/DefaultConsistentHashFactoryTest.java

```diff
@@ -207,16 +209,16 @@ protected void checkDistribution(DefaultConsistentHash ch, Map<Address, Float> l
          int minPrimaryOwned = (int) Math.floor(minOwned(numSegments, 1, numNodesWithLoad, expectedPrimaryOwned));
          int maxPrimaryOwned = (int) Math.ceil(maxOwned(numSegments, 1, numNodesWithLoad, expectedPrimaryOwned));
          int primaryOwned = stats.getPrimaryOwned(node);
-         assertTrue(minPrimaryOwned <= primaryOwned && primaryOwned <= maxPrimaryOwned,
-                    String.format("Primary owned (%d) should have been between %d and %d",
-                                  primaryOwned, minPrimaryOwned, maxPrimaryOwned));
+         assert minPrimaryOwned <= primaryOwned && primaryOwned <= maxPrimaryOwned :
```

Comment:
why replacing  `assertTrue` with `assert` keyword?

Reply:
Performance: the `String.format()` call was slowing down the test.

Reply:
does it matter for testing?

Reply:
Replacing all the assert keywords with assertTrue makes the test approx. 500ms slower for on my machine, and it runs twice in the suite. It may not sound much, but not worrying about the test duration got us to waiting at least 2 hours for a PR's tests.

I ran the test with even more segments/nodes combinations locally while testing, so IMO it does matter.

Reply:
500 ms? damn 
my only concern is if disable assertions on the jvm by mistake; then the test would worthless. 
can you try using `if`?
```java
if (minPrimaryOwned > primaryOwned || primaryOwned > maxPrimaryOwned)
    fail(String.format("Primary owned (%d) should have been between %d and %d", primaryOwned, minPrimaryOwned, maxPrimaryOwned));
```

Reply:
The test does a lot of iterations :)

In the end I applied your suggestion and I also removed some of the rebalance iterations (e.g. no more rebalancing with null capacity factors, since they're the same as cf(X)=1 for each X), which shaved about 2 seconds more.

Reply:
> In the end I applied your suggestion and I also removed some of the rebalance iterations (e.g. no more rebalancing with null capacity factors, since they're the same as cf(X)=1 for each X), which shaved about 2 seconds more.

@danberindei have you pushed those changes yet?

Reply:
I don't know if I had pushed them when when you asked, but they're in now.